ruby/day10/codesmell/order_processor.rb

```diff
@@ -0,0 +1,45 @@
+require './customer'
+require './item'
+require './order'
+
+class OrderProcessor
+  def process_order(order)
+    @customer = order.customer
+    @items = order.items
+
+    raise "Không có mặt hàng nào!" if @items.empty?
+    
+    # Kiểm tra tồn kho
+    check_quantity
+
+    # Tạo hóa đơn
+    invoice = create_invoice(get_price - add_discount)
+
+    # Gửi email xác nhận
+    send_confirmation_email(invoice)
+  end
+  
+  def get_price              
+    total_price = 0
+    @items.each do |item|
+      total_price += item.price * item.quantity

```

Comment:
ở đây đang sử dụng price và quantity của item để tính toán (feature envy) --> đưa vào 1 method tính price ở Item class